### PR TITLE
Add a new variant dimension for HVR packages in China

### PR DESCRIPTION
**Pull request by svillar**
The mainland China release of HVR packages will have some specific configurations. We have been using different branches for that but it's unmanageable. Instead we should add a new flavor dimension for mainland China and the rest of the world.

The first difference we'll have is that the tutorial in the phone UI won't be shown for mainland China releases. The package names will be different too due to particularities of the Huawei store.

This requires a sequence of commits. First we refactor the current variant filtering. Then we add the new flavor dimension. Finally after setting an specific application id for China, we can disable showing the phone UI on that country.

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -634,6 +626,25 @@ if (gradle.hasProperty('localProperties.dependencySubstitutions.geckoviewTopsrcd
     apply from: "${topsrcdir}/substitute-local-geckoview.gradle"
 }
 
+androidComponents {
+    onVariants(selector().all(), { variant ->
+        switch (variant.name) {
+            case "hvr3dofArm64CnDebug":
+                variant.applicationId = "cn.igalia.wolvic.tdof." + getDevApplicationIdSuffix()
```
Comment (felipeerias) [inline app/build.gradle:633]: This fails to compile if `getDevApplicationIdSuffix()` returns an empty string.

File: app/build.gradle [hunk 2]
```diff
@@ -251,7 +251,6 @@ android {
             }
             buildConfigField "String", "HVR_ML_API_KEY", "\"${getHVRMLApiKey()}\""
             buildConfigField "String", "MK_API_KEY", "\"\""
-            buildConfigField "Boolean", "WEBVIEW_IN_PHONE_UI", "true"
```
Comment (felipeerias) [inline app/build.gradle:254]: I suggest to keep this build parameter and use something like this to change its value:

```js
    applicationVariants.all { variant ->
        def platform = variant.productFlavors.get(0).name
        def country = variant.productFlavors.get(2).name

        if (platform.toLowerCase().startsWith('hvr') && country == 'cn')
            variant.buildConfigField "Boolean", "WEBVIEW_IN_PHONE_UI", "false"
    }
```

Comment (svillar) [reply app/build.gradle:254]: OK

File: app/build.gradle (hunk 1 above)
Comment (svillar) [reply app/build.gradle:633]: Ah because of the leading dot. Good catch

File: app/build.gradle [hunk 3]
```diff
@@ -653,4 +657,9 @@ android.applicationVariants.all { variant ->
         println("X_X")
     }
 
+    def platform = variant.productFlavors.get(0).name
+    def country = variant.productFlavors.get(2).name
+
+    if (platform.toLowerCase().startsWith('hvr') && country != 'cn')
+        variant.buildConfigField "Boolean", "WEBVIEW_IN_PHONE_UI", "true"
```
[the remark this answers is not in the document]
Comment (svillar) [reply app/build.gradle:664]: Good catch. That was a mistake in the rebase.

**Pull request merged**